grpc/pb-marshalling.go

```diff
@@ -360,10 +360,68 @@ func pbToAuthz(pb *rapb.Authorization) (core.Authorization, error) {
 	}, nil
 }
 
-func registrationValid(reg *rapb.Registration) bool {
+func registrationValid(reg *corepb.Registration) bool {
 	return !(reg.Id == nil || reg.Key == nil || reg.Agreement == nil || reg.InitialIP == nil || reg.CreatedAt == nil || reg.Status == nil || reg.ContactsPresent == nil)
 }
 
-func authorizationValid(authz *rapb.Authorization) bool {
+func authorizationValid(authz *corepb.Authorization) bool {
 	return !(authz.Id == nil || authz.Identifier == nil || authz.RegistrationID == nil || authz.Status == nil || authz.Expires == nil)
 }
+
+func certificateValid(cert *corepb.Certificate) bool {
+	return !(cert.RegistrationID == nil || cert.Serial == nil || cert.Digest == nil || cert.Der == nil || cert.Issued == nil || cert.Expires == nil)
+}
+
+func sctToPB(sct core.SignedCertificateTimestamp) *sapb.SignedCertificateTimestamp {
+	id := int64(sct.ID)
+	version := int64(sct.SCTVersion)
+	timestamp := int64(sct.Timestamp)
+	return &sapb.SignedCertificateTimestamp{
+		Id:                &id,
+		SctVersion:        &version,
+		LogID:             &sct.LogID,
+		Timestamp:         &timestamp,
+		Extensions:        sct.Extensions,
+		Signature:         sct.Signature,
+		CertificateSerial: &sct.CertificateSerial,
+	}
+}
+
+func pbToSCT(pb *sapb.SignedCertificateTimestamp) core.SignedCertificateTimestamp {
+	return core.SignedCertificateTimestamp{
+		ID:                int(*pb.Id),
```

Comment:
Ah, we do that in [`sctToPB`](https://github.com/letsencrypt/boulder/pull/2374/files/ceee39a941e9f847ebccd6ec1302a350a50f1dcd#diff-f88663c9cfc4ccabf0b33c739f9f8435R375), this is converting back from the protobuf type so needs to cast to the native types we use in `core.SignedCertificateTimestamp` or it would fail.